stdlib/universe/fill.gen.go

```diff
@@ -0,0 +1,221 @@
+// Generated by tmpl
+// https://github.com/benbjohnson/tmpl
+//
+// DO NOT EDIT!
+// Source: fill.gen.go.tmpl
+
+package universe
+
+import (
+	"fmt"
+
+	"github.com/apache/arrow/go/arrow"
+	"github.com/apache/arrow/go/arrow/array"
+	"github.com/influxdata/flux"
+)
+
+func (t *fillTransformation) fillColumn(typ flux.ColType, arr array.Interface) array.Interface {
+	switch typ {
+	case flux.TInt:
+		return t.fillIntColumn(arr.(*array.Int64))
+	case flux.TUInt:
+		return t.fillUintColumn(arr.(*array.Uint64))
+	case flux.TFloat:
+		return t.fillFloatColumn(arr.(*array.Float64))
+	case flux.TBool:
+		return t.fillBooleanColumn(arr.(*array.Boolean))
+	case flux.TString:
+		return t.fillStringColumn(arr.(*array.Binary))
+	case flux.TTime:
+		return t.fillTimeColumn(arr.(*array.Int64))
+
+	default:
+		panic(fmt.Errorf("unsupported array data type: %s", arr.DataType()))
+	}
+}
+
+func (t *fillTransformation) fillIntColumn(arr *array.Int64) array.Interface {
+	var prevValue int64
```

Comment:
I think this won't work correctly with multiple buffers. We had this problem with derivative. The previous value has to be kept outside of the `Do` loop so it will need to be passed in.

I'd suggest changing the structure to this:

```
func (t *fillTransformation) fillColumn(typ flux.ColType, arr array.Interface, v *interface{}) {
    // previous code
}

func (t *fillTransformation) fillIntColumn(arr *array.Int64, v *interface{}) array.Interface {
    var prevValue int64
    prevNull := *v == nil
    if !prevNull {
        prevValue = (*v).(int64)
    }
    if t.spec.UsePrevious {
        // Save the previous value for this buffer to use in future buffers.
        defer func() {
            if !prevNull { *v = prevValue }
        }()
    }
}

// not valid code but a rough idea of what it would look like
func (t *fillTransformation) Process(tbl flux.Table) error {
    var fillValue interface{}
    if !t.spec.Value.IsNull() {
        fillValue = /* extract the value from the value */
    }
    tbl.Do(func(cr flux.ColReader) error {
        t.fillColumn(typ, arr, &fillValue)
    })
}
```

I prefer to avoid `values.Value` if possible for this lower level code. I don't think there's any difference if you replaced `interface{}` with `values.Value` so feel free to replace the `interface{}` uses with `values.Value` if you find that easier.